src/core/lib/iomgr/exec_ctx.cc

```diff
@@ -38,26 +74,21 @@ bool grpc_exec_ctx_ready_to_finish(grpc_exec_ctx* exec_ctx) {
   }
 }
 
-bool grpc_never_ready_to_finish(grpc_exec_ctx* exec_ctx, void* arg_ignored) {
-  return false;
-}
+bool grpc_never_ready_to_finish(void* arg_ignored) { return false; }
 
-bool grpc_always_ready_to_finish(grpc_exec_ctx* exec_ctx, void* arg_ignored) {
-  return true;
-}
+bool grpc_always_ready_to_finish(void* arg_ignored) { return true; }
 
-bool grpc_exec_ctx_has_work(grpc_exec_ctx* exec_ctx) {
+bool grpc_exec_ctx_has_work() {
   return exec_ctx->active_combiner != NULL ||
          !grpc_closure_list_empty(exec_ctx->closure_list);
 }
 
-void grpc_exec_ctx_finish(grpc_exec_ctx* exec_ctx) {
+void grpc_exec_ctx_finish() {
   exec_ctx->flags |= GRPC_EXEC_CTX_FLAG_IS_FINISHED;
-  grpc_exec_ctx_flush(exec_ctx);
+  grpc_exec_ctx_flush();
 }
 
-static void exec_ctx_run(grpc_exec_ctx* exec_ctx, grpc_closure* closure,
-                         grpc_error* error) {
+static void exec_ctx_run(grpc_closure* closure, grpc_error* error) {
```

Comment:
Should be a private member function on ExecCtx... `Run(grpc_closure*, grpc_error*)`